iidm/iidm-xml-converter/src/main/java/com/powsybl/iidm/xml/ShuntXml.java

```diff
@@ -134,8 +149,11 @@ protected void readElement(String id, ShuntCompensatorAdder adder, NetworkXmlRea
             adder.setSectionCount(sectionCount);
         });
         readNodeOrBus(adder, context);
-        double p = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "p");
-        double q = XmlUtil.readOptionalDoubleAttribute(context.getReader(), "q");
+
+        return voltageRegulatorOn[0];
+    }
+
+    private ShuntCompensator readSubElementsShuntCompensator(String id, ShuntCompensatorAdder adder, NetworkXmlReaderContext context) throws XMLStreamException {
```

Comment:
I think you can call override the `readSubElements` method with this one and call that rather than creating another

Reply:
Finally,  the standard structure of reading has been preserved.